src/Illuminate/Cache/FileStore.php

```diff
@@ -111,7 +111,9 @@ public function put($key, $value, $minutes)
     protected function createCacheDirectory($path)
     {
         try {
-            $this->files->makeDirectory(dirname($path), 0777, true, true);
+            if (! file_exists(dirname($path))) {
```

Comment:
Not sure you understand what I meant? This class can't know about the file system. You have to make that check through another class that does.